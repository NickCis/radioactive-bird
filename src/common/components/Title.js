import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

const Title = ({ title, children }) => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) staticContext.title = title;
      else document.title = title;
      return children || null;
    }}
  />
);

Title.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default Title;
