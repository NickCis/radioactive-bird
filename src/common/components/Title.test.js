import React from 'react';
import Title from './Title';
import { StaticRouter, MemoryRouter } from 'react-router';
import { mount } from 'enzyme';

describe('<Titile />', () => {
  it('should set the title of the static context', () => {
    const context = {};
    const title = 'test title';

    mount(
      <StaticRouter location="/" context={context}>
        <Title title={title} />
      </StaticRouter>
    );

    expect(context.title).toBe(title);
  });

  it('should set the title of document', () => {
    const title = 'test title';

    mount(
      <MemoryRouter>
        <Title title={title} />
      </MemoryRouter>
    );

    expect(document.title).toBe(title);
  });

  it('should render childs', () => {
    const context = {};
    const title = 'test title';

    const wrapper = mount(
      <StaticRouter location="/" context={context}>
        <Title title={title}>
          <div>This is a test</div>
        </Title>
      </StaticRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
