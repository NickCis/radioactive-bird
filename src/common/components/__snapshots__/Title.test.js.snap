// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Titile /> should render childs 1`] = `
<StaticRouter
  basename=""
  context={
    Object {
      "title": "test title",
    }
  }
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Title
      title="test title"
    >
      <Route
        render={[Function]}
      >
        <div>
          This is a test
        </div>
      </Route>
    </Title>
  </Router>
</StaticRouter>
`;
